File: modules/privileges.py
# This Python file uses the following encoding: utf-8

# MODULES AND/OR LIBRARIES
from sys import exit as sysexit
from os import geteuid
from modules.logging_config import(
    debug,
    info,
)

##############################

# ROOT PRIVILEGES

##############################

def check_root_privileges():
    effective_user_id = geteuid()
    debug(f"Effective User ID: {effective_user_id}") 
    if effective_user_id != 0:
        print("[!] This scan requires root privileges.")
        info("Exiting the program due to lack of root privileges.")
        sysexit(1)

File: modules/privileges.go
package modules

import (
	"fmt"
	"os"
	"syscall"
)

// ##############################
//
// # ROOT PRIVILEGES
//
// ##############################

func CheckRootPrivileges() {
	euid := syscall.Geteuid()
	fmt.Printf("[DEBUG] Effective User ID: %d\n", euid)

	if euid != 0 {
		fmt.Println("[!] This scan requires root privileges.")
		fmt.Println("[INFO] Exiting the program due to lack of root privileges.")
		os.Exit(1)
	}
}